### ISSUE TITLE
seaf-daemon occupies CPU

### ISSUE THREAD

**Issue by DevCybran** [haiwen/seafile-client#604]
The process seaf-daemon.exe has one thread running at 100% CPU usage when continously writing to a file inside a seafile library.

This issue occured when I downloaded multiple large files using firefox into a seafile-monitored directory.

**Comment by shoeper** [comment]
That's normal behaviour. On detecting a change Seafile directly indexes data and uploads it. Indexing data costs cpu.

**Comment by mnott** [comment]
Nice. But should it really keep above 60 % like /all the time/???

Over here, it is on Yosemite, according to the process panel, responsible for consuming 70 % of battery.

There should be a option to throttle it.

**Comment by shoeper** [comment]
No. Only when data is changing. But of course, the behaviour could get improved.

**Comment by mnott** [comment]
Well, in that case it is like quite difficult to "tame" Seafile. It would be great having some kind of statistics to see which amount of time it spent on which library or subfolder, for indexing. With that, one could identify potentially bad candidates.

**Comment by inbaz** [comment]
I'm on Windows 7 and encounter the same problem. Sometimes the seaf-daemon.exe takes 75% CPU constantly. What information is needed to solve this issue?

**Comment by shoeper** [comment]
It is likely that one of your libraries is constantly getting changed at that time.

**Comment by FeuRenard** [comment]
I use the seafile-client under Fedora 24. The seaf-daemon process periodically occupies 25% of my CPU (which means 100% of one core for a hyperthreaded dual core CPU). In the applet then for at least one library an exclamation mark symbol is shown, which shows two different mouse over texts:

- failed to get sync info from server
- sync initializing

seafile.log shows this lines, many times repeated. (xxx in the URLs added by me)
> [07/24/16 13:35:39] http-tx-mgr.c(947): libcurl failed to POST https://login.yoursecurecloud.de/seafhttp/repo/folder-perm: SSL connect error.
> [07/24/16 13:35:56] http-tx-mgr.c(707): libcurl failed to GET https://login.yoursecurecloud.de/seafhttp/repo/xxx/commit/HEAD: SSL connect error.
> [07/24/16 13:35:56] http-tx-mgr.c(707): libcurl failed to GET https://login.yoursecurecloud.de/seafhttp/repo/xxx/commit/HEAD: SSL connect error.
> [07/24/16 13:35:56] http-tx-mgr.c(707): libcurl failed to GET https://login.yoursecurecloud.de/seafhttp/repo/xxx/commit/HEAD: SSL connect error.
> [07/24/16 13:35:56] sync-mgr.c(728): Repo 'Dokumente' sync state transition from initializing to 'error': 'Failed to get sync info from server.'.
> [07/24/16 13:35:56] sync-mgr.c(728): Repo 'Fw-Archiv' sync state transition from initializing to 'error': 'Failed to get sync info from server.'.
> [07/24/16 13:35:56] sync-mgr.c(728): Repo 'Bilder' sync state transition from initializing to 'error': 'Failed to get sync info from server.'.

**Comment by xkpd3** [comment]
I just found this issue. I have the same issue since a long time. Deleting all repositories and resyncing sometimes helps.

I'm in a situation where everything behaves normal. But the CPU is at 100% without any activity (all files synced). I noticed that there is a number on constant VCSW (voluntary context switches) and IVCSW (involuntary context switches) going on. I'm not sure what this means, but it is not expected :)

I'm running the seafile-applet 6.1.1 on FreeBSD 12-CURRENT (same issue is happening on 11-RELEASE).

Screenshot with Log (top terminal), top -mio (like iotop on linux, middle terminal), top (bottom terminal), seafile window (right side).

![Screenshot](https://shell.srv.hagen.coffee/~sdk/screenshots/2017-10-07_21-29-45.png)

**Comment by mky** [comment]
I also have this issue. The seaf-daemon uses 100% CPU all the time. When I delete my Libraries locally and sync it again, it helps for some time but after some time, the issue appear again. I am using Seafile-applet 6.1.1 on FreeBSD 11.1-RELEASE. Any solution for this?

**Comment by xkpd3** [comment]
This is still open and still an issue. I've recently update my seafile server and client. The behavior is still the same.

Unlike in the linked issue, the 100% CPU is not related to the network connectivity. I don't see any unusual log entries.

I'm happy to help debugging this issue if someone can walk me though on IRC or the like. (I've never debugged python).

**Comment by xkpd3** [comment]
I played around with dtrace. While seaf-daemon should be idle and all Libraries are synced, dtrace reports that more than 100K select:entry syscalls are fired within 5 seconds:
```
root@ptrace:~ # dtrace -n 'syscall:freebsd:select:entry /execname == "seaf-daemon"/ { @ = count (); } tick-5s { exit(0); }'
dtrace: description 'syscall:freebsd:select:entry ' matched 2 probes
CPU     ID                    FUNCTION:NAME
  0  81415                         :tick-5s

           100291
```
